### PR TITLE
HPCC-20017 Unable to review packagemap info

### PR DESCRIPTION
**Pull request by miguelvazq**
Grid rendering was not occuring due to a grid.store.query is not found. This is a legacy widget that still needs some updating. The grid now populates with packages and the user can continue onto review packagemap info.

Signed-off-by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by richardkchapman** [comment]
@miguelvazq

**Comment by miguelvazq** [comment]
@GordonSmith updated.

File: esp/src/src/ESPPackageProcess.ts [hunk 1]
```diff
@@ -1,135 +1,21 @@
...
+
+var Store = declare([ESPRequest.Store], {
+    i18n: nlsHPCC,
+    service: "WsPackageProcess",
+    action: "ListPackages",
+    responseQualifier: "ListPackagesResponse.PackageMapList.PackageListMapData",
+    responseTotalQualifier: "ListPackagesResponse.PackageMapList.PackageListMapData",
+    idProperty: "__hpcc_id"
+    // startProperty: "PageStartFrom",
+    // countProperty: "PageSize",
```
Comment (GordonSmith) [inline esp/src/src/ESPPackageProcess.ts]: Remove commented code

File: esp/src/src/ESPPackageProcess.ts [hunk 2]
```diff
@@ -1,135 +1,21 @@
...
-    })
-});
-
 export function CreatePackageMapQueryObjectStore(options) {
     var store = new Store(options);
-    store = Observable(store);
-    var objStore = new ObjectStore({ objectStore: store });
-    return objStore;
-}
+    return new Observable(store);
```
Comment (GordonSmith) [inline esp/src/src/ESPPackageProcess.ts]: I don't think it matters, but I think its meant to be:  `return Observable(store);`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith updated.

File: esp/src/eclwatch/TargetSelectClass.js [hunk 3]
```diff
@@ -169,10 +175,38 @@ define([
...
                     }
+                }).then(function (response) {
+                    if (lang.exists("GetPackageMapSelectOptionsResponse.Targets.TargetData", response)) {
+                        var targetData = response.GetPackageMapSelectOptionsResponse.Targets.TargetData;
+                        context.options.push({
+                            label: "ANY",
+                            value: "*"
+                        });
+                        for (var i = 0; i < targetData.length; ++i) {
+                            if (targetData[i].Processes.Item) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/TargetSelectClass.js]: Should this be `lang.exists("Processes.Item.length", targetData[i])`?

**Review by GordonSmith — CHANGES_REQUESTED** [review]
One minor comment - otherwise looks good

**Comment by ghalliday** [comment]
@miguelvazq

**Comment by miguelvazq** [comment]
@GordonSmith updated and tested.

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Comment by miguelvazq** [comment]
@ghalliday ready for merge.

**Pull request merged**